app/src/main/java/com/concordia/mcga/models/Campus.java

```diff
@@ -73,26 +73,28 @@ public String getShortName() {
     }
 
     /**
-     * @param polygon polygon object to search for
-     * @return return the building which contains this polygon
+     *
+     * @param object can be a marker object or polygon object to search for
+     * @return return the building which contains this marker or polygon
      */
-    public Building getBuilding(Polygon polygon){
-        for (Building building : buildings) {
-            if(building.getPolygon().getId().equalsIgnoreCase(polygon.getId()))
-                return building;
+    public static Building getBuilding(Object object) {
```

Comment:
I prefer overloading the method rather than making the parameter a generic Object